R/loadFromMetaphlan.R

```diff
@@ -0,0 +1,188 @@
+#'
+#' @param sample_meta a single \code{character} value defining the file
+#'   path of the sample metadata file. The file must be in \code{tsv} format
+#'   (default: \code{sample_meta = NULL}).
+#'   
+#' @param phy_tree a single \code{character} value defining the file
+#'   path of the phylogenetic tree.
+#'   (default: \code{phy_tree = NULL}).
+#'   
+#' @param ... additional arguments:
+#' \itemize{
+#'   \item{\code{removeTaxaPrefixes}:} {\code{TRUE} or \code{FALSE}: Should
+#'     taxonomic prefixes be removed? (default:
+#'     \code{removeTaxaPrefixes = FALSE})}
+#' }
+#'
+#' @details
+#' Import Metaphlan results. Input must be in merged Metaphlan format.
+#' Data is imported so that data at the lowest rank is imported as a 
+#' \code{TreeSE} object. Data at higher rank is imported as a
+#' \code{SE} objects which are stored to \code{altExp} of
+#' \code{TreeSE} object.
+#'
+#' @return  A
+#' \code{\link[TreeSummarizedExperiment:TreeSummarizedExperiment-class]{TreeSummarizedExperiment}}
+#' object
+#'
+#' @name loadFromMetaphlan
+#' @seealso
+#' \code{\link[=makeTreeSummarizedExperimentFromPhyloseq]{makeTreeSummarizedExperimentFromPhyloseq}}
+#' \code{\link[=makeSummarizedExperimentFromBiom]{makeSummarizedExperimentFromBiom}}
+#' \code{\link[=makeTreeSummarizedExperimentFromDADA2]{makeTreeSummarizedExperimentFromDADA2}}
+#' \code{\link[=loadFromQIIME2]{loadFromQIIME2}}
+#' \code{\link[=loadFromMothur]{loadFromMothur}}
+#'
+#' @export
+#' @author Leo Lahti and Tuomas Borman. Contact: \url{microbiome.github.io}
+#' 
+#' @references
+#'
+#' @examples
+#' \dontrun{
+#' # File path
+#' file_path <- "/data/metaphlan_result.txt"
+#' # Import data
+#' tse <- loadFromMetaphlan(file_path)
+#' # Data at the lowest rank
+#' tse
+#' # Data at higher rank is stored in altExp
+#' altExps(tse)
+#' # Higher rank data is in SE format, for example, Phylum rank
+#' altExp(tse, "Phylum")
+#' }
+#' 
+
+loadFromMetaphlan <- function(metaphlan, sample_meta = NULL, phy_tree = NULL, ...){
+    ################################ Input check ################################
+    if(!.is_non_empty_string(metaphlan)){
+        stop("'metaphlan' must be a single character value.",
+             call. = FALSE)
+    }
+    if (!file.exists(metaphlan)) {
+        stop(metaphlan, " does not exist", call. = FALSE)
+    }
+    if(!is.null(sample_meta) && !.is_non_empty_string(sample_meta)){
+        stop("'sample_meta' must be a single character value or NULL.",
+             call. = FALSE)
+    }
+    if(!is.null(phy_tree) && !.is_non_empty_string(phy_tree)){
+        stop("'phy_tree' must be a single character value or NULL.",
+             call. = FALSE)
+    }
+    ############################## Input check end #############################
+    # Create a list SE objects from the Metaphlan data. All represent own taxonomic rank
+    se_objects <- .read_metaphlan_into_se_objects(metaphlan, ...)
```

Comment:
I would structure it differently. This has to much nesting.

How about this:
```r
.read_metaphlan(metaphlan, ...)
.check_metaphlan_data(...)
.parse_metaphlan_data(...)
.create_se_from_metaphlan(...)
```

Reply:
Bumping this. it is shown as outdated but it isn't

Reply:
Now I restructured it

Also, I added `abund_values` for selecting the name of assay (hidden with` ...` )